karoo_gp/engine.py

```diff
@@ -0,0 +1,184 @@
+import os
+import ast
+import operator as op
+
+import numpy as np
+
+from . import Tree
+from .util import LazyLoader
+# Tensorflow takes ~2 seconds to load, so only load when used
+tf = LazyLoader('tf', globals(), 'tensorflow.compat.v1')
+
+
+class Engine:
```

Comment:
Is this supposed to be an abstract base class?
If so, it needs a few changes.

Reply:
Lets discuss